lib/generators/shopify_app/install/templates/content_security_policy.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+# Be sure to restart your server when you modify this file.
+
+# Define an application-wide content security policy
+# For further information see the following documentation
+# https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy
+
+# Rails.application.configure do
+#   config.content_security_policy do |policy|
+#     policy.default_src :self, :https
+#     policy.font_src    :self, :https, :data
+#     policy.img_src     :self, :https, :data
+#     policy.object_src  :none
+#     policy.script_src  :self, :https
+#     policy.style_src   :self, :https
+#     # Specify frame ancestors for embedded apps
+#     # For more information, see the following documentation
+#     # https://shopify.dev/apps/store/security/iframe-protection
+#     policy.frame_ancestors :https, -> { "https://#{current_shopify_domain} https://admin.shopify.com" }
```

Comment:
How is the `current_shopify_domain` retrieve from the session? We should provide the example. 

Also, in a case that `current_shopify_domain` is `nil` (pre-auth) we should fallback to `[:https, "*.myshopify.com", "admin.shopify.com".

Reply:
Instead of checking for `nil` you will need to check for `defined?` and I totally agree with you that there has to be a fallback.

Reply:
Maybe I'm wrong with this and please correct me if I am as this issue seems to concern a lot of public app developers using this gem.

In the Rails documentation it says (see: https://guides.rubyonrails.org/security.html#content-security-policy-header) 

> Use lambdas to inject per-request values, such as account subdomains in a multi-tenant application:

with an example of

```ruby
class PostsController < ApplicationController
  content_security_policy do |policy|
    policy.base_uri :self, -> { "https://#{current_user.domain}.example.com" }
  end
end
```

shouldn't that be the approach of this PR? I just see a lot of problems by having it as an initialized file where the context is not always present and it will also be called on requests to the application that might not need this policy at all (e.g. if you make a request to the application from the storefront instead it will break since `current_shopify_domain` will never be defined).

Looking at how this gem operates currently I would assume a controller concern that uses a similar way as this https://github.com/Shopify/shopify_app/blob/e5350a587594169b3506c8e63356df0844dabc7b/app/controllers/concerns/shopify_app/require_known_shop.rb#L12-L15 to fetch the shop domain would be more appropriate (and also not dependent on the `RequireKnownShop` concern).

I would really appreciate some communication on this.